### PR TITLE
Lit update

### PR DESCRIPTION
**Pull request by Doxxmedearly**
Lit cigarettes now count as flame sources.
Changed the isflamesource proc from a general proc to an obj/item proc. It is tidier and will be easier to add future flame sources this way. 
Removed some hardcoded flame stuff (Such as igniting rags and paper), which depending on an item being a type instead of just being something that could produce a flame. They now check to see if an item counts as a flame source.
Found leftover clown BS while updating paper burning code and killed it off.

### CODE REVIEW

**Comment by Doxxmedearly** [comment]
!review

File: code/modules/detectivework/tools/rag.dm [hunk 1]
```diff
@@ -53,14 +53,12 @@
...
-		if(F.lit)
-			ignite()
-			if(on_fire)
-				visible_message(span("warning", "\The [user] lights \the [src] with \the [W]."))
-			else
-				to_chat(user, span("warning", "You manage to singe \the [src], but fail to light it."))
+	if(!on_fire && W.isFlameSource())
+		ignite()
+		if(on_fire)
+			visible_message(span("warning", "\The [user] lights \the [src] with \the [W]."))
```
Comment (Geevies) [inline code/modules/detectivework/tools/rag.dm:59]:
```suggestion
			visible_message(SPAN_WARNING("\The [user] lights \the [src] with \the [W]."))
```

File: code/modules/detectivework/tools/rag.dm [hunk 2]
```diff
@@ -53,14 +53,12 @@
...
+		else
+			to_chat(user, span("warning", "You manage to singe \the [src], but fail to light it."))
```
Comment (Geevies) [inline code/modules/detectivework/tools/rag.dm:61]:
```suggestion
			to_chat(user, SPAN_WARNING(You manage to singe \the [src], but fail to light it."))
```

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 3]
```diff
@@ -103,6 +91,7 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 	var/icon_off
 	var/type_butt = null
 	var/chem_volume = 15 //Size of a syringe
+	var/genericmes = "USER lights NAME with FLAME"
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/cigs_lighters.dm:94]: wayment, what's the point of this?

Comment (Doxxmedearly) [reply code/game/objects/items/weapons/cigs_lighters.dm:94]: Cigarettes have different vars to describe different messages when lighting a cigarette with a flame source. This is why lighting something with an igniter has a different visible_message than a zippo, has a different visible_message than a welder, etc.
`if(istype(W, /obj/item/flame/match))
			text = matchmes
		else if(istype(W, /obj/item/flame/lighter/zippo))
			text = zippomes
		else if(istype(W, /obj/item/flame/lighter))
			text = lightermes
		else if(W.iswelder())
			text = weldermes
		else if(istype(W, /obj/item/device/assembly/igniter))
			text = ignitermes`

Genericmes was added as an else as a catch-all to anything that didn't fall under these typepaths so a visible_message would be displayed.

Comment (Doxxmedearly) [reply code/game/objects/items/weapons/cigs_lighters.dm:94]: I guess matchmes could be used as the generic but this way we could give matches unique messages in the future.

**Pull request merged**